x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java

```diff
@@ -196,6 +198,37 @@ public DocValuesScriptFieldFactory getScriptFieldFactory(String name) {
                             throw new UnsupportedOperationException("The [" + CONTENT_TYPE + "] field does not " + "support scripts");
                         }
 
+                        @Override
+                        public FormattedDocValues getFormattedValues(DocValueFormat format) {
+                            try {
+                                final BinaryDocValues values = DocValues.getBinary(context.reader(), fieldName);
+                                final InternalHistogramValue value = new InternalHistogramValue();
+                                return new FormattedDocValues() {
+                                    @Override
+                                    public boolean advanceExact(int docId) throws IOException {
+                                        return values.advanceExact(docId);
+                                    }
+
+                                    @Override
+                                    public int docValueCount() {
+                                        return 1;
```

Comment:
I am not sure if this is correct if we have an empty field. Will addd a test.

Reply:
Note that here I am assuming that given a document and a field in that document we only have one histogram and that we don't have a situation where we have a field whit an array of histograms as value.

Reply:
I think this is fine, because the caller excepts a `InternalHistogramValue` instance and only one will be emitted per document.

Reply:
> I am assuming that given a document and a field in that document we only have one histogram and that we don't have a situation where we have a field with an array of histograms as value.

Histogram fields are always single-value. Your assumption is correct